sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/InvocationVisitor.cs

```diff
@@ -160,18 +184,21 @@ private ProgramState HandleInstanceEqualsCall()
                 .PopValues(validatedArgumentIndex)
                 .SetConstraint(guardedArgumentValue, ObjectConstraint.NotNull);
 
-        private static readonly ISet<string> IsNullMethodNames = new HashSet<string>
-        {
-            nameof(string.IsNullOrEmpty),
-            nameof(string.IsNullOrWhiteSpace),
-        };
 
         private static bool IsStringNullCheckMethod(IMethodSymbol methodSymbol)
         {
             return methodSymbol != null &&
                 methodSymbol.ContainingType.Is(KnownType.System_String) &&
                 methodSymbol.IsStatic &&
-                IsNullMethodNames.Contains(methodSymbol.Name);
+                nameof(string.IsNullOrWhiteSpace) == methodSymbol.Name;
+        }
+
+        private static bool IsStringNullOrEmptyCheckMethod(IMethodSymbol methodSymbol)
+        {
+            return methodSymbol != null &&

```

Comment:
you could extract common behavor in `IsStringStaticMethod(IMethodSymbol, string methodName)` and also use arrow notation